email_account.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"database/sql"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	emailLib "github.com/jordan-wright/email"
+)
+
+type EmailAccount struct {
+	Id        string    `json:"id,omitempty"`
+	Email     string    `json:"email"`
+	PubKey    string    `json:"pubkey,omitempty"`
+	HasPubKey bool      `json:"has_pubkey,omitempty"`
+	Created   time.Time `json:"created,omitempty"`
+}
+
+func GetEmailAccount(id string, db *sql.DB) (*EmailAccount, error) {
+	emailAccount := &EmailAccount{}
+	err := db.QueryRow(`
+		SELECT
+			id, email, has_pubkey, created
+		FROM
+			email_account
+		WHERE
+		id = $1
+	`, id).Scan(
+		&emailAccount.Id, &emailAccount.Email,
+		&emailAccount.HasPubKey, &emailAccount.Created,
+	)
+
+	if err != nil {
+		if err != sql.ErrNoRows {
+			log.Errorf("Error getting email_account. Err: %s", err)
+		}
+		return nil, err
+	}
+
+	return emailAccount, nil
+}
```

Comment:
@ajvb How about implementing this as just a special case of `GetEmailAccounts`, where you pass in `[]string{id}` and return `accounts[0]`?

Reply:
Personally, I'm not a fan of that idea. I think creating special cases within function are usually hard to understand and fragile, where the current implementation is clear/readable.

Reply:
Oh I misunderstood what you were staying completely! I like it.